pandas/core/generic.py

```diff
@@ -4516,6 +4516,9 @@ def describe_1d(data, percentiles):
                 if name not in names:
                     names.append(name)
         d = pd.concat(ldesc, join_axes=pd.Index([names]), axis=1)
+        # if hierarchical index has column names, add column names to d.
+        if self.columns.names:
+            d.columns.names = self.columns.names
```

Comment:
This should always be done (no need for the if)

further, should be ``d.columns.names = data.columns.names``